go/adbc/driver/snowflake/bulk_ingestion.go

```diff
@@ -595,3 +678,39 @@ func (bp *bufferPool) PutBuffer(buf *bytes.Buffer) {
 	buf.Reset()
 	bp.Pool.Put(buf)
 }
+
+type fileSet struct {
+	mu   sync.RWMutex
+	data map[string]struct{}
+}
```

Comment:
is this better / worse than just using `sync.Map`? 

`sync.Map` is generally best for primarily append workflows with few delete/removes, but for our workflows here I doubt it would make too much of a difference other than avoiding us having to maintain this?

Reply:
I actually started with `sync.Map` but changed to `map` + `sync.RMutex` because I thought it was causing a bug. Turned out to be something else and `sync.Map` works just fine.

I just pushed a commit. I left it wrapped in the `fileSet` struct because using a naked `sync.Map` and having to remember to use the basename everywhere + writing the `Len()` iteration inline feels a little awkward IMHO. Let me know what you think.

Reply:
I changed this to a type definition `type fileSet sync.Map`.